src/stats.c

```diff
@@ -689,8 +672,12 @@ static ALWAYS_INLINE UInt ExecWhileHelper(Stat stat, UInt nr)
     body2 = (nr >= 2) ? READ_STAT(stat, 2) : 0;
     body3 = (nr >= 3) ? READ_STAT(stat, 3) : 0;
 
+    // HACK: point current "statement" at the condition expression, so
+    // that if there is an error in it, it resp. its location is printed,
+    // not the whole loop statement.
+    SET_BRK_CALL_TO(cond);
```

Comment:
We might want to add a few more of these "hacks" for now. The purpose of these hacks is as follows: Without them, you might see:
```
Error, Rational operations: <divisor> must not be zero in
  if true = 1 / 0 then
    return 1;
fi; at *stdin*:9 called from
```
with the hacks, you get:
```
Error, Rational operations: <divisor> must not be zero in
  1 / 0 at *stdin*:9 called from
```
I.e. it's the difference between printing the condition expression of a loop or `if` clause, vs. printing the whole loop resp. `if` statement. Now, a better approach probably would be to either print the actual line(s) containing the expression (as e.g. GDB would do it); or else add a way to print just the "head" of a loop or `if` (resp.: print the loop, but with the body replaced by, say `...`). The former can be difficult because a single expression can of course span many lines, including comments; also, the source file may be modified, leading to output that's not right (just as can happen with gdb); if we wanted to avoid that, we'd have to store the whole file in memory.

The second approach seems much simpler to me, and shouldn't be too hard to implement... (famous last words)

Reply:
BTW, these HACKS are also limited in so far as they break down if `cond` is an "immediate" expression, i.e., an `INTEXPR` or a `REFLVAR`

I'll look into implementing the second approach now